pkg/api/v1beta1/clustercurator_types.go

```diff
@@ -60,6 +60,28 @@ type Hooks struct {
 	OverrideJob *runtime.RawExtension `json:"overrideJob,omitempty"`
 }
 
+type UpgradeHooks struct {
+
+	// DesiredUpdate indicates the desired value of
+	// the cluster version. Setting this value will trigger an upgrade (if
+	// the current version does not match the desired version).
+	// +optional
+	DesiredUpdate string `json:"desiredUpdate,omitempty"`
+
+	// Channel is an identifier for explicitly requesting that a non-default
+	// set of updates be applied to this cluster. The default channel will be
+	// contain stable updates that are appropriate for production clusters.
+	// +optional
+	Channel string `json:"channel,omitempty"`
+
+	// Upstream may be used to specify the preferred update server. By default
+	// it will use the appropriate update server for the cluster and region.
+	// +optional
+	Upstream string `json:"upstream,omitempty"`
```

Comment:
not sure about the `appropriate update server` . is it always a fixed server for default?

Reply:
I assume this text is just borrowed from the cluster version CRD.